### PR TITLE
[SYCL] Add definition of CL_SYCL_LANGUAGE_VERSION for SYCL-2020

### PR DESCRIPTION
**Pull request by s-kanaev**

### CODE REVIEW

File: clang/lib/Frontend/InitPreprocessor.cpp [hunk 1]
```diff
@@ -467,6 +467,7 @@ static void InitializeStandardPredefinedMacros(const TargetInfo &TI,
       Builder.defineMacro("CL_SYCL_LANGUAGE_VERSION", "121");
       Builder.defineMacro("SYCL_LANGUAGE_VERSION", "201707");
     } else if (LangOpts.SYCLVersion == 2020) {
+      Builder.defineMacro("CL_SYCL_LANGUAGE_VERSION", "2020");
```
Comment (s-kanaev) [inline clang/lib/Frontend/InitPreprocessor.cpp:470]: I'm not sure if this is by-spec.

Comment (Fznamznon) [reply clang/lib/Frontend/InitPreprocessor.cpp:470]: SYCL 1.2.1 defines `CL_SYCL_LANGUAGE_VERSION`, whereas SYCL 2020 provisional defines `SYCL_LANGUAGE_VERSION`. I'm not sure that this is correct.

Comment (elizabethandrews) [reply clang/lib/Frontend/InitPreprocessor.cpp:470]: The change looks ok to me but I don't know what the value for CL_SYCL_LANGUAGE_VERSION should be. @Fznamznon would you know?

**Review by elizabethandrews — COMMENTED** [review]
Please also add a test

File: clang/lib/Frontend/InitPreprocessor.cpp (hunk 1 above)
Comment (elizabethandrews) [reply clang/lib/Frontend/InitPreprocessor.cpp:470]: Oops. I see you commented right before I did.

> SYCL 1.2.1 defines `CL_SYCL_LANGUAGE_VERSION`, whereas SYCL 2020 provisional defines `SYCL_LANGUAGE_VERSION`. I'm not sure that this is correct.

SYCL 1.2.1  defines `CL_SYCL_LANGUAGE_VERSION` and `SYCL_LANGUAGE_VERSION`.

**Comment by s-kanaev** [comment]
Closing as suggested changes are not by-spec.

**Pull request closed without merging**